File: src/api/test_user_action.py
from fastapi.testclient import TestClient
from user_action import app

test_client = TestClient(app)

def test_log_user_action():
    response = test_client.post('/user-action', json={'action': 'jump', 'timestamp': '2023-01-01T00:00:00Z'})
    assert response.status_code == 200
    assert response.json() == {'message': 'User action logged', 'action': 'jump'}

File: src/api/test_user_action.js
const request = require('supertest');
const app = require('../server');

describe('POST /api/user-action', () => {
    it('should record a user action successfully', async () => {
        const response = await request(app)
            .post('/api/user-action')
            .send({ action: 'start_game', userId: 'user123' });
        expect(response.status).toBe(201);
        expect(response.body.message).toBe('User action recorded.');
    });

    it('should return error for missing action', async () => {
        const response = await request(app)
            .post('/api/user-action')
            .send({ userId: 'user123' });
        expect(response.status).toBe(400);
        expect(response.body.message).toBe('Action and User ID are required.');
    });
});